pie/trainer.py

```diff
@@ -355,7 +436,6 @@ def train_epochs(self, epochs, devset=None):
                          "task [{}] with best score {:.4f}".format(e.task, e.loss))
 
             self.model.load_state_dict(e.best_state_dict)
-            scores = {e.task: e.loss}
```

Comment:
Unless I am missing something, this change would break the report for best loss in the target task in the case of early stopping. The EarlyStopException is raised by train epoch, and the best loss might come from previous epochs.

Reply:
Well, technically, it was use nowhere from what I saw... Hence me removing it. Initially, I commented it out but I ended up removing it.

Reply:
It is used right below. It's the output of the method. It used to be used to write the output path for the model but now not anymore. But in any case I think it should be there regardless of what happens in train.py

Reply:
It's used below but wasn't use elsewhere. What I'll probably do is simply update the dictionary with (scores is already a dict of task, i'll just add loss) or return a tuple.

Reply:
If training exits through early stopping the scores will be wrong. What's returned through the exception is the target task and the metric being monitored ('accuracy', 'all') by default. If you don't use that information (which isn't complete) at least you should rerun dev evaluation in the except block.

Reply:
That makes total sense actually... I'll do it ! 
Have a nice week end ;)